packages/schematics/angular/migrations/use-application-builder/migration.ts

```diff
@@ -362,10 +363,21 @@ export default function (): Rule {
     ),
     // Update main tsconfig
     updateJsonFile('tsconfig.json', (rootJson) => {
-      rootJson.modify(['compilerOptions', 'esModuleInterop'], true);
+      let hasPreserveModule = false;
+      const compilerOptions = rootJson.get(['compilerOptions']);
+
+      if (compilerOptions && typeof compilerOptions === 'object') {
+        const { module } = compilerOptions as JsonObject;
+        hasPreserveModule = typeof module === 'string' && module.toLowerCase() === 'preserve';
+      }
```

Comment:
This can be simplified a bit.

```suggestion
   const module = rootJson.get(['compilerOptions', 'module']);
   const hasPreserveModule = typeof module === 'string' && module.toLowerCase() === 'preserve';
```